src/test/lib/BasicTest.cpp

```diff
@@ -270,3 +270,26 @@ void QuicTestBindConnectionExplicit(_In_ int Family)
         TEST_QUIC_SUCCEEDED(Status);
     }
 }
+
+void QuicTestAddrFunctions(_In_ int Family)
+{
+    QUIC_ADDR SockAddr = {};
+    QUIC_ADDRESS_FAMILY QuicAddrFamily = (Family == 4) ? QUIC_ADDRESS_FAMILY_INET : QUIC_ADDRESS_FAMILY_INET6;
+    QuicAddrSetFamily(&SockAddr, QuicAddrFamily);
+    TEST_TRUE(QuicAddrGetFamily(&SockAddr) == QuicAddrFamily);
+
+    if (QuicAddrFamily == QUIC_ADDRESS_FAMILY_INET) {
+        SockAddr.Ipv4.sin_addr.s_addr = 0x00FFFF00UL;
+        QuicAddrSetToLoopback(&SockAddr);
+        TEST_TRUE(SockAddr.Ipv4.sin_addr.s_host == 0);
```

Comment:
I _think_ most of the direct access you're doing for the inner variables of the IP address won't work cross platform.